.gitlab-ci.yml

```diff
@@ -1,8 +1,22 @@
+before_script:
+  - >
+    if echo "$CI_COMMIT_TITLE" | grep -q "ci: Update snapshots"; then
```

Comment:
Actually I think we can do something better: can we define a generic prefix, for instance `ci-skip-rust:` and `ci-skip-e2e:` which can be followed by a generic string indicating the reason, that would let us skip either the Rust or the e2e tests? So we can push different commit "semantics" that would all have the same effect, i.e., skipping some tests.

Reply:
Anyway if it's not easily doable, don't spend too much time on this. We will overhaul the CI structure anyway at some point.

Reply:
If this does what's intended, I think it's good 👍 Question: where can we document this, for us and for others?

Reply:
It should work. I have to example commits where you can investigate the output: 

for rust: [ba85c34](https://github.com/KILTprotocol/kilt-node/pull/639/commits/ba85c34538b4f401584bebd772e4593e8980eb6d) pipeline: https://gitlab.com/kiltprotocol/kilt-node/-/pipelines/1295088348 

for integration tests: [f5449e7](https://github.com/KILTprotocol/kilt-node/pull/639/commits/f5449e7ea432c3cb0a5791e3c55dbb117a73d54d) pipeline: https://gitlab.com/kiltprotocol/kilt-node/-/pipelines/1295116631

Reply:
I'm unsure where to place the documentation. I'm considering introducing a `WORKFLOW_README.md` in the root folder. We could also add an explanation to the [command-bot](https://github.com/KILTprotocol/command-bot), but I'm not certain if that's the right place.

Reply:
Yeah it does not look like we can add it to the README. Maybe we can use a "Contributing" file like the one the polkadot-sdk repo has: https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/CONTRIBUTING.md.

Reply:
For me, it feels like a new issue. I created a new ticket to track this: [#3416](https://github.com/KILTprotocol/ticket/issues/3416). We should also maybe discuss which aspects should be described.